src/main/java/com/bamboo/tloll/collision/CollisionDetector.java

```diff
@@ -0,0 +1,113 @@
+package com.bamboo.tloll.collision;
+
+import com.bamboo.tloll.graphics.Unit;
+import com.bamboo.tloll.graphics.structure.Tile;
+import com.bamboo.tloll.graphics.structure.WorldMap;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
+
+public class CollisionDetector
+{
+
+    private static CollisionDetector _instance;
+
+    public static CollisionDetector getInstance() {
+        if (_instance == null) {
+            _instance = new CollisionDetector();
+        }
+        return _instance;
+    }
+
+    private CollisionDetector() {
+    }
+    
+    public List<Tile> getOccupiedTiles(Unit player)
+    {
+        WorldMap wm = WorldMap.getInstance();
```

Comment:
I think the two instances of these should be refactored into an overloaded method. So getOccupiedTiles(Unit player)
should just call getOccupiedTiles(Unit player, float deltaX, float deltaY) and send in 0 for deltaX and deltaY. It appears they are identical aside from that.